### PR TITLE
Front currency

### PR DESCRIPTION
**Pull request by MartynasKasp**

### CODE REVIEW

File: client/src/containers/currencySubscriptions/form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          <Button className={style.Form__Button} color="primary" onClick={handleToList}>
+            Back to list
+          </Button>
+        </div>
+      </div>
+    </Container>
+  );
+};
+
+export default CurrencySubscriptionForm;
```
Comment (MantasGra) [inline client/src/containers/currencySubscriptions/form/index.tsx]: nėra endline, turbūt nepralintinta

File: client/src/containers/currencySubscriptions/form/style.module.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.Form__Container {
```
Comment (MantasGra) [inline client/src/containers/currencySubscriptions/form/style.module.scss:1]: įdomus naming

File: client/src/containers/currencySubscriptions/list/index.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        className={style.fab}
+        onClick={handleCreate}
+      >
+        <AddIcon />
+      </Fab>
+    </Container>
+  );
+};
+
+export default CurrencySubscriptions;
```
Comment (MantasGra) [inline client/src/containers/currencySubscriptions/list/index.tsx]: same as above.

**Comment by MartynasKasp** [comment]
solid work fam

File: client/src/store/modules/currencySubscriptions/reducer.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      return {
+        ...state,
+        errors: { currency: '' },
+      };
+    default:
+      return state;
+  }
+};
+
+export default reducer;
```
Comment (MantasGra) [inline client/src/store/modules/currencySubscriptions/reducer.ts]: same as above

File: client/src/store/modules/currencySubscriptions/requests.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export const createCurrencySubscription = (
+  data: CurrencySubscriptionFormType
+) => {
+  return axios.post('https://localhost:5001/api/currency-subscriptions', {
+    ...data,
+    currency: CurrencySubscriptionCurrencies[data.currency],
+    userId: 1,
+  })
+    .then((res: AxiosResponse) => res.data);
+};
```
Comment (MantasGra) [inline client/src/store/modules/currencySubscriptions/requests.ts]: same as above.

File: client/src/store/modules/currencySubscriptions/sagas.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+function* rootSaga(){
+  yield all([
+    fork(getCurrencySubscriptionsWatcher),
+    fork(createCurrencySubscriptionWatcher),
+  ]);
+}
+
+export default rootSaga;
```
Comment (MantasGra) [inline client/src/store/modules/currencySubscriptions/sagas.ts]: same as above

File: client/src/store/types.ts [hunk 7]
```diff
@@ -1,9 +1,11 @@
...
 import { TransactionState } from './modules/transactions/types';
+import { CurrencySubscriptionsState } from './modules/currencySubscriptions';
 
 export interface State {
   transactions: TransactionState;
   account: AccountsState;
   globals: GlobalState;
-}
+  currencySubscription: CurrencySubscriptionsState;
+}
```
Comment (MantasGra) [inline client/src/store/types.ts]: same as above.

File: client/src/utils/types.ts [hunk 8]
```diff
@@ -1 +1,2 @@
 export type Dictionary<T> = { [id: string]: T };
+
```
Comment (MantasGra) [inline client/src/utils/types.ts]: actually good change xD

**Review by MantasGra — APPROVED** [review]
good stuff

File: client/src/containers/currencySubscriptions/form/index.tsx (hunk 1 above)
Comment (MartynasKasp) [reply client/src/containers/currencySubscriptions/form/index.tsx]: gali būti

File: client/src/containers/currencySubscriptions/form/style.module.scss (hunk 2 above)
Comment (MartynasKasp) [reply client/src/containers/currencySubscriptions/form/style.module.scss:1]: BEM convension guy

File: client/src/containers/currencySubscriptions/form/index.tsx (hunk 1 above)
Comment (MartynasKasp) [reply client/src/containers/currencySubscriptions/form/index.tsx]: jo buvo nepralintinta

File: client/src/containers/currencySubscriptions/form/style.module.scss (hunk 2 above)
Comment (MantasGra) [reply client/src/containers/currencySubscriptions/form/style.module.scss:1]: aha tik tu vienas naudoji tokį convention šitam projekte xD

**Pull request merged**